src/sidebar/components/svg-icon.js

```diff
@@ -1,34 +1,32 @@
 'use strict';
 
-/**
- * The <svg-icon> component renders SVG icons using inline <svg> tags,
- * enabling their appearance to be customized via CSS.
- *
- * This matches the way we do icons on the website, see
- * https://github.com/hypothesis/h/pull/3675
- */
+const { createElement } = require('preact');
+const propTypes = require('prop-types');
 
 // The list of supported icons
 const icons = {
   refresh: require('../../images/icons/refresh.svg'),
   cursor: require('../../images/icons/cursor.svg'),
 };
 
-// @ngInject
-function SvgIconController($element) {
-  this.$onInit = () => {
-    if (!icons[this.name]) {
-      throw new Error('Unknown icon: ' + this.name);
-    }
-    $element[0].innerHTML = icons[this.name];
-  };
+/**
+ * Component that renders icons using inline `<svg>` elements.
+ * This enables their appearance to be customized via CSS.
+ *
+ * This matches the way we do icons on the website, see
+ * https://github.com/hypothesis/h/pull/3675
+ */
+function SvgIcon({ name }) {
+  if (!icons[name]) {
+    throw new Error(`Unknown icon ${name}`);
+  }
+  const markup = { __html: icons[name] };
+  return <span dangerouslySetInnerHTML={markup} />;
```

Comment:
I notice this adds a span. Wondering if there is a way to do with without an additional span element?

Reply:
Good question. The extra element is only here temporarily. Currently the DOM structure in the Angular version looks like this:

```html
<svg-icon ...>
  <svg>...</svg>
</svg-icon>
```

On this branch an additional `<span>` is added so it will look like this:

```html
<svg-icon ...>
  <span><svg>...</svg></span>
</svg-icon>
```

After the Angular component containing the `<svg-icon>` is converted however, the outer element will disappear, since React components don't appear in the DOM tree directly, only their output:

```html
<span><svg>...</svg></span>
```

Rendering just the `<svg>` here would be possible, but it would require additional code to parse the markup and extract the outer `<svg>` element. I avoided that to keep it simpler.

Reply:
It might help to add a comment about this, and/or to track this so we get back to it—it does seem ideal to "fix" this at some point.